From 2a06d4fcdee63610d408bdf5a04bd502ec448338 Mon Sep 17 00:00:00 2001
From: Sergei Shkonda <sergeish@marvell.com>
Date: Wed, 1 Mar 2017 17:14:29 +0200
Subject: [PATCH 1036/1345] dts: msys: Enable Switching Core interrupt on AC3

commit  633350b0eec3cadcc8816eaa755128d6cea64dca from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

    Change pinctrl: "mpp13" function must be "pp"
    Switching core interrupt is 33

Change-Id: I747fa19bb44628847ae8461e7ea0f50ff6142619
Reviewed-on: http://vgitil04.il.marvell.com:8080/39578
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Terry Zhou <bjzhou@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/msys-ac3.dtsi |    5 +++++
 arch/arm/boot/dts/msys.dtsi     |    7 ++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/msys-ac3.dtsi b/arch/arm/boot/dts/msys-ac3.dtsi
index dc83c9c..108d17d 100644
--- a/arch/arm/boot/dts/msys-ac3.dtsi
+++ b/arch/arm/boot/dts/msys-ac3.dtsi
@@ -21,5 +21,10 @@
 				compatible = "marvell,ac3-pinctrl";
 			};
 		};
+		prestera {
+			pinctrl-names = "default";
+			pinctrl-0 = <&pp_pin>;
+			interrupts = <33>;
+		};
 	};
 };
diff --git a/arch/arm/boot/dts/msys.dtsi b/arch/arm/boot/dts/msys.dtsi
index 36722a7..0a8288d 100644
--- a/arch/arm/boot/dts/msys.dtsi
+++ b/arch/arm/boot/dts/msys.dtsi
@@ -116,7 +116,7 @@
 									"mpp3", "mpp5", "mpp6",
 									"mpp7", "mpp8", "mpp9",
 									"mpp10", "mpp11", "mpp12",
-									"mpp13", "mpp16", "mpp17",
+									"mpp16", "mpp17",
 									"mpp20", "mpp21", "mpp22",
 									"mpp23", "mpp24", "mpp25",
 									"mpp26", "mpp27", "mpp28",
@@ -150,6 +150,11 @@
 					marvell,function = "uart1";
 				};
 
+				pp_pin: pp-pin {
+					marvell,pins = "mpp13";
+					marvell,function = "pp";
+				};
+
 				i2c0_pins: i2c0-pins {
 					marvell,pins = "mpp14", "mpp15";
 					marvell,function = "i2c0";
-- 
1.7.9.5

